### PR TITLE
ref(server): Internal Arc for EnvelopeProccesorService

### PR DESCRIPTION
**Pull request by TBS1996**
This refactor is needed for #2320, made it a separate PR to make it easier to read the diffs on that branch

We want to make envelopeprocessorservice easily cloneable, so we put most of the data it contains behind an arc. This allows us to remove the outer arc in the spawn_handler, so that we can mutate the service. Which will be needed for global config update.

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/processor.rs [hunk 1]
```diff
@@ -520,8 +520,13 @@ impl FromMessage<RateLimitFlushBuckets> for EnvelopeProcessor {
 /// Service implementing the [`EnvelopeProcessor`] interface.
 ///
 /// This service handles messages in a worker pool with configurable concurrency.
+#[derive(Clone)]
 pub struct EnvelopeProcessorService {
     config: Arc<Config>,
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs]: `config` can also move into `inner`. Only the new global config (which is not part of this PR) will be outside of the `Arc`.

File: relay-server/src/actors/processor.rs [hunk 2]
```diff
@@ -541,6 +546,7 @@ impl EnvelopeProcessorService {
         project_cache: Addr<ProjectCache>,
         upstream_relay: Addr<UpstreamRelay>,
     ) -> Self {
+        // TODO(tor): Replace with state enum (`init`, `ready`) and do not process anything while global_config is undefined.
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs]: Does this comment make sense in this PR?
```suggestion
```

File: relay-server/src/actors/processor.rs [hunk 3]
```diff
@@ -520,8 +520,13 @@ impl FromMessage<RateLimitFlushBuckets> for EnvelopeProcessor {
 /// Service implementing the [`EnvelopeProcessor`] interface.
 ///
 /// This service handles messages in a worker pool with configurable concurrency.
+#[derive(Clone)]
 pub struct EnvelopeProcessorService {
     config: Arc<Config>,
+    inner: Arc<InnerEnvelopeProcessorService>,
+}
+
+struct InnerEnvelopeProcessorService {
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs]: nit: Since this is a private type, I would be fine with a shorter name, e.g. `InnerProcessor`.

File: relay-general/src/store/normalize/span/tag_extraction.rs
Comment (jjbayer) [inline relay-general/src/store/normalize/span/tag_extraction.rs:1]: Changes in this file are unrelated.

**Review by jjbayer — APPROVED** [review]
See comments.

File: relay-server/src/actors/processor.rs (hunk 1 above)
Comment (TBS1996) [reply relay-server/src/actors/processor.rs]: I wasn't sure why but this is what jan was suggesting when all paired 🤔

File: relay-general/src/store/normalize/span/tag_extraction.rs
Comment (TBS1996) [reply relay-general/src/store/normalize/span/tag_extraction.rs:1]: true but my linter complained about them so might as well change it now. seems it suddenly got the ability to spot unneeded clones

File: relay-server/src/actors/processor.rs (hunk 1 above)
Comment (TBS1996) [reply relay-server/src/actors/processor.rs]: ok nvm, was a misunderstanding, changed it now

**Comment by jjbayer** [comment]
Also please capitalize the PR title 😄
